02_classify_mutations/nbconverted/new_vs_old_genes.py

```diff
@@ -0,0 +1,244 @@
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+# location of classification results
+results_dir = Path(cfg.results_dirs['mutation'], 'merged_all', 'gene').resolve()
+
+# set significance cutoff after FDR correction
+SIG_ALPHA = 0.001
+
+
+# ### Load results and perform statistical testing/correction
+
+# In[3]:
+
+
+# load raw data
+results_df = au.load_stratified_prediction_results(results_dir, 'gene')
+
+# here we want to use compressed data for methylation datasets (27k and 450k)
+# the results in 02_classify_compressed/compressed_vs_raw_results.ipynb show that
+# performance is equal or slightly better for PCA compressed methylation data,
+# and it's much easier/faster to fit models on
+results_df = results_df[results_df.training_data.isin(['expression', 'rppa', 'mirna', 'mut_sigs'])]
+
+# make sure that we're correctly pointing to raw data for non-methylation data types
+# and that we have data for two replicates (two random seeds)
+print(results_df.shape)
+print(results_df.seed.unique())
+print(results_df.training_data.unique())
+results_df.head()
+
+
+# In[4]:
+
+
+# load compressed data for me_27k and me_450k
+compressed_results_df = au.load_compressed_prediction_results(results_dir, 'gene')
+
+# make sure that we're correctly pointing to compressed methylation data
+# and that we have data for one dimension and two replicates (two random seeds)
+print(compressed_results_df.shape)
+print(compressed_results_df.seed.unique())
+print(compressed_results_df.training_data.unique())
+print(compressed_results_df.n_dims.unique())
+compressed_results_df.head()
+
+
+# In[5]:
+
+
+results_df['n_dims'] = 'raw'
+results_df = pd.concat((results_df, compressed_results_df))
+print(results_df.seed.unique())
+print(results_df.training_data.unique())
+print(results_df.n_dims.unique())
+print(results_df.shape)
+training_data_map = {
+    'expression': 'gene expression',
+    'me_27k': '27k methylation',
+    'me_450k': '450k methylation',
+    'rppa': 'RPPA',
+    'mirna': 'microRNA',
+    'mut_sigs': 'mutational signatures',
+}
+results_df.training_data.replace(to_replace=training_data_map, inplace=True)
+results_df.head()
+
+
+# In[6]:
+
+
+all_results_df = au.compare_all_data_types(results_df,
+                                           SIG_ALPHA,
+                                           metric='aupr')
+
+cfg.sig_genes_dir.mkdir(exist_ok=True)
+all_results_df.to_csv(cfg.sig_genes_all, index=False, sep='\t')
+
+all_results_df.sort_values(by='p_value').head(10)
+
+
+# ### Load gene sets
+
+# In[7]:
+
+
+# load datasets
+vogelstein_df = du.load_vogelstein()
+vogelstein_df.head()
+
+
+# In[8]:
+
+
+# load datasets
+merged_df = du.load_merged()
+merged_df.head()
+
+
+# In[9]:
+
+
+vogelstein_genes = set(vogelstein_df.gene)
+non_vogelstein_genes = set(merged_df.gene) - vogelstein_genes
+print(len(vogelstein_genes), len(non_vogelstein_genes))
+
+
+# ### Plot results for Vogelstein and non-Vogelstein genes separately
```

Comment:
Is this what you mean by new vs old genes? That's not said explicitly anywhere.

Reply:
I'll add a comment at the top of the script!